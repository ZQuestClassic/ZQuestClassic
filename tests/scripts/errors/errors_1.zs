#option NO_ERROR_HALT on

global script Global
{
    void missing_run()
    {
        int number = 123;
        printf("hello world: %d\n", number_oops);
    }
}
