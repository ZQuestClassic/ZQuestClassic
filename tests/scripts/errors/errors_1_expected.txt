stderr:

Compiling 'errors_1.zs'
Pass 1: Parsing
Pass 2: Preprocessing
Pass 3: Registration
Pass 4: Analyzing Code
errors_1.zs Line 3 @ Columns 15-21 - Error S011: Script Global needs a void run() function.

1    #option NO_ERROR_HALT on
2    
3    global script Global
                   ^~~~~~


errors_1.zs Line 8 @ Columns 37-48 - Error S009: Variable number_oops has not been declared.

6    {
7        int number = 123;
8        printf("hello world: %d\n", number_oops);
                                     ^~~~~~~~~~~


Failure!
Compile finished with exit code '11' (compiled with errors)

stdout:

{
  "success": false,
  "code": 11,
  "diagnostics": [
    {
      "range": {
        "start": {
          "line": 2,
          "character": 14
        },
        "end": {
          "line": 2,
          "character": 20
        }
      },
      "severity": 1,
      "message": "S011: Script Global needs a void run() function."
    },
    {
      "range": {
        "start": {
          "line": 7,
          "character": 36
        },
        "end": {
          "line": 7,
          "character": 47
        }
      },
      "severity": 1,
      "message": "S009: Variable number_oops has not been declared."
    }
  ],
  "metadata": {
    "currentFileSymbols": 3,
    "symbols": 0,
    "identifiers": 0,
    "elided": true
  }
}
